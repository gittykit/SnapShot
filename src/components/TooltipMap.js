import React, { Component } from "react";
import RCTooltip from 'rc-tooltip'

import Map from './Map'

class TooltipMap extends Component {
  render() {
    const {children, lat, lng, title} = this.props
    const coords = {
      lat: Number(lat),
      lng: Number(lng)
    }

    const mapProps = {
      options: {
        center: coords,
        zoom: 10,
        disableDefaultUI: true,
      },
      onMount: this.getMarker(coords, title)
    }

    const tooltipContent= (
      <div className="tooltipMap-wrapper">
        <div className='tooltipMap-title'>{title}</div>
        <Map className="tooltipMap-map" {...mapProps} />
      </div>
    )

    return (
      <RCTooltip
        overlay={tooltipContent}
        destroyTooltipOnHide={false}
        mouseEnterDelay={0.1}
        mouseLeaveDelay={0.1}
        placement={'top'}
      >
        {children}
      </RCTooltip>
    );
  }

  getMarker(coords, title) {
    return (map) => {
      new window.google.maps.Marker({
        map,
        position: coords,
        label: '',
        title: title,
      });
    };
  }
}

export default TooltipMap;
