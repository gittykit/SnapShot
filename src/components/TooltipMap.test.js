import React from "react";
import { shallow } from 'enzyme'
import TooltipMap from "./TooltipMap"

describe('TooltipMap', () => {
  it("renders", () => {
    const children = (<span>test</span>)
    const defaultProps = {
      lat: 23,
      lng: 20,
      title: "titletest",
      children
    }
    const wrapper = shallow(<TooltipMap {...defaultProps} />)

    expect(wrapper).toMatchSnapshot()
  });
})
