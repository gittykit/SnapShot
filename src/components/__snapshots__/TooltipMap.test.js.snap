// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipMap renders 1`] = `
<ForwardRef(Tooltip)
  destroyTooltipOnHide={false}
  mouseEnterDelay={0.1}
  mouseLeaveDelay={0.1}
  overlay={
    <div
      className="tooltipMap-wrapper"
    >
      <div
        className="tooltipMap-title"
      >
        titletest
      </div>
      <Map
        className="tooltipMap-map"
        onMount={[Function]}
        options={
          Object {
            "center": Object {
              "lat": 23,
              "lng": 20,
            },
            "disableDefaultUI": true,
            "zoom": 10,
          }
        }
      />
    </div>
  }
  placement="top"
>
  <span>
    test
  </span>
</ForwardRef(Tooltip)>
`;
